pallets/dkg-metadata/src/lib.rs

```diff
@@ -737,150 +740,172 @@ impl<T: Config> Pallet<T> {
+			if !signer.can_sign() {
+				return Err(
+					"No local accounts available. Consider adding one via `author_insertKey` RPC.",
+				)
+			}
 
-		if let Ok(Some(agg_keys)) = agg_keys {
-			let _res = signer.send_signed_transaction(|_account| Call::submit_next_public_key {
-				keys_and_signatures: agg_keys.clone(),
-			});
+			if let Ok(Some(agg_keys)) = agg_keys {
+				let _res = signer.send_signed_transaction(|_account| {
+					Call::submit_next_public_key { keys_and_signatures: agg_keys.clone() }
+				});
 
-			agg_key_ref.clear();
-		}
+				agg_key_ref.clear();
+			}
 
-		Ok(())
+			Ok(())
+		}
 	}
 
 	fn submit_public_key_signature_onchain(
 		_block_number: T::BlockNumber,
 	) -> Result<(), &'static str> {
-		let mut pub_key_sig_ref = StorageValueRef::persistent(OFFCHAIN_PUBLIC_KEY_SIG);
+		let mut lock = StorageLock::<Time>::new(b"submit_public_key_signature_onchain::lock");
+		{
+			let _guard = lock.lock();
 
-		if Self::next_public_key_signature().is_some() {
-			pub_key_sig_ref.clear();
-			return Ok(())
-		}
+			let mut pub_key_sig_ref = StorageValueRef::persistent(OFFCHAIN_PUBLIC_KEY_SIG);
 
-		let refresh_proposal = pub_key_sig_ref.get::<RefreshProposalSigned>();
+			if Self::next_public_key_signature().is_some() {
+				pub_key_sig_ref.clear();
+				return Ok(())
+			}
 
-		let signer = Signer::<T, T::OffChainAuthId>::all_accounts();
-		if !signer.can_sign() {
-			return Err(
-				"No local accounts available. Consider adding one via `author_insertKey` RPC.",
-			)
-		}
+			let refresh_proposal = pub_key_sig_ref.get::<RefreshProposalSigned>();
 
-		if let Ok(Some(refresh_proposal)) = refresh_proposal {
-			let _ = signer.send_signed_transaction(|_account| Call::submit_public_key_signature {
-				signature_proposal: refresh_proposal.clone(),
-			});
-			frame_support::log::debug!(target: "dkg", "Offchain submitting public key sig onchain {:?}", refresh_proposal.signature);
+			let signer = Signer::<T, T::OffChainAuthId>::all_accounts();
+			if !signer.can_sign() {
+				return Err(
+					"No local accounts available. Consider adding one via `author_insertKey` RPC.",
+				)
+			}
 
-			pub_key_sig_ref.clear();
-		}
+			if let Ok(Some(refresh_proposal)) = refresh_proposal {
+				let _ =
+					signer.send_signed_transaction(|_account| Call::submit_public_key_signature {
+						signature_proposal: refresh_proposal.clone(),
+					});
+				frame_support::log::debug!(target: "dkg", "Offchain submitting public key sig onchain {:?}", refresh_proposal.signature);
+
+				pub_key_sig_ref.clear();
+			}
 
-		Ok(())
+			Ok(())
+		}
 	}
 
 	fn submit_misbehaviour_reports_onchain(
 		_block_number: T::BlockNumber,
 	) -> Result<(), &'static str> {
-		let signer = Signer::<T, T::OffChainAuthId>::all_accounts();
-		if !signer.can_sign() {
-			return Err(
-				"No local accounts available. Consider adding one via `author_insertKey` RPC.",
-			)
-		}
+		let mut lock = StorageLock::<Time>::new(b"submit_public_key_signature_onchain::lock");
```

Comment:
Same here too:
> Could be these keys here `b"submit_genesis_public_key_onchain::lock"` be constants, similar to `AGGREGATED_PUBLIC_KEYS_AT_GENESIS` for example? To avoid typos.